src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
```

Comment:
Partially correct. To optimize this for maps with multiple types of aircraft that can land on carriers you would want to sort air units by 'defense value divided by carrier cost' to get the best air units per capacity (in most situations). 

This still doesn't guarantee the best though since you could then have situations where ignoring the best air unit per capacity allows more units to fit. Only way to truly make sure is try all possible combinations of air units which could be costly if there are say 10-20 carriers with lots of different air unit combinations. It also makes this method way more complex and would take a good deal of time to determine what is the best way to handle it. 

Summary: Most maps only have 1 type of fighter or the types of fighters are so similar that it doesn't make much of a difference (Global 40). So for this bug fix I decided to keep it simple and just randomly choose the planes that end up not fitting on the carriers.